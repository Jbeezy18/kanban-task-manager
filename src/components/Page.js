import * as React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setTitle } from "../actions/pageActions";

class Page extends React.Component{

    constructor(props) {
        super(props);
        this.changeTitle = this.changeTitle.bind(this);
    }

    changeTitle(){
        console.log("changeTitle");
        let newTitle = document.getElementById("title").value;
        console.log(newTitle);
        this.props.setTitle(newTitle);
    }

    render() {
        return (
            <div>
                <h1>{this.props.title}</h1>
                <input id={"title"} type="text"/>
                <button onClick={this.changeTitle}>Change Title</button>
            </div>
        );
    }
}

Page.propTypes ={
  setTitle: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
   title: state.title.title
});

export default connect(mapStateToProps, { setTitle })(Page);