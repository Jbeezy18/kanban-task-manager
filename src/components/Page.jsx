import * as React from 'react';

export default class Page extends React.Component{
    render() {
        return (
            <div>
                <h1>{this.props.title}</h1>
                <button>Nothing will happen!</button>
            </div>
        );
    }
}
