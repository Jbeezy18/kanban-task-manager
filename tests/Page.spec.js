import React from 'react';
import renderer from 'react-test-renderer';

import Page from "../src/components/Page";

describe('Page', () =>{
    test('Page snapshot renders', () => {
        const tree = renderer
            .create(<Page title={"Hi Long!"}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});