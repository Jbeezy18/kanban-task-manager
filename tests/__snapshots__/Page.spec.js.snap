// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Page snapshot renders 1`] = `
<div>
  <h1>
    Hi Bitches!
  </h1>
  <button>
    Nothing will happen!
  </button>
</div>
`;
